xbmc/network/NetworkServices.cpp

```diff
@@ -826,10 +826,16 @@ bool CNetworkServices::StartUPnP()
 {
   bool ret = false;
 #ifdef HAS_UPNP
-  ret |= StartUPnPClient();
-  ret |= StartUPnPServer();
-  ret |= StartUPnPController();
-  ret |= StartUPnPRenderer();
+  if (CSettings::GetInstance().GetBool(CSettings::SETTING_SERVICES_UPNPSERVER)) {
+	  ret |= StartUPnPClient();
```

Comment:
The client should always be started because it is required for browsing UPnP sources independent of whether the user wants to share its library or not.